internal/jobs/launch_instance_aws.go

```diff
@@ -144,27 +144,37 @@ func DoEnsurePubkeyOnAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) e
 	fingerprint := pubkey.FindAwsFingerprint(ctx)
 	ec2Name, err := ec2Client.GetPubkeyName(ctx, fingerprint)
 	if err != nil {
-		span.SetStatus(codes.Error, "key error")
+		logger.Trace().Msgf("Pubkey with fingerprint '%s' not found, importing type %s key '%s'", fingerprint, pubkey.Type, pubkey.Body)
 
 		// if not found on AWS, import
 		if errors.Is(err, http.PubkeyNotFoundErr) {
 			pkr.Tag = ""
 			pkr.RandomizeTag()
+			retryCount := 1
+			origName := pubkey.Name
+		retry:
 			pkr.Handle, err = ec2Client.ImportPubkey(ctx, pubkey, pkr.FormattedTag())
 
 			if errors.Is(err, http.DuplicatePubkeyErr) {
+				retryCount += 1
+				pubkey.Name = fmt.Sprintf("%s%d", origName, retryCount)
+				if retryCount < 4 {
```

Comment:
I feel like this might get confusing. This says that the retryCount is < 4, but this retry block is actually performed twice. This is just by the fact that we are starting from 1 and also checking for "<".
Just by the looks of it, I would expect the ImportPubkey to be invoked at least three times. Can we change this a bit so it is more clear how many times we actually run this?

Reply:
This was not intentional, you are right. Rebased a change with starting from zero.